larpandoracontent/LArTwoDReco/LArClusterAssociation/CrossGapsExtensionAlgorithm.cc

```diff
@@ -51,72 +51,90 @@ void CrossGapsExtensionAlgorithm::GetListOfCleanClusters(const ClusterList *cons
 
 void CrossGapsExtensionAlgorithm::FillClusterAssociationMatrix(const ClusterVector &clusterVector, ClusterAssociationMatrix &clusterAssociationMatrix) const
 {
-    // Build lists of pointing clusters in proximity to gaps
-    LArPointingClusterList innerPointingClusterList, outerPointingClusterList;
-    this->BuildPointingClusterList(clusterVector, innerPointingClusterList, outerPointingClusterList);
+    // Build and loop over list of pointing clusters
+    LArPointingClusterList pointingClusterList;
+    this->BuildPointingClusterList(clusterVector, pointingClusterList);
 
-    // Form associations between pairs of pointing clusters
-    for (const LArPointingCluster &pointingClusterInner : innerPointingClusterList)
+    for (LArPointingClusterList::const_iterator iter1 = pointingClusterList.begin(), iterEnd1 = pointingClusterList.end();
+        iter1 != iterEnd1; ++iter1)
```

Comment:
Could now use to c++11 range-based for loop, if you want. This looks clunky now! Ditto below.

Reply:
That's a good point. But these aren't full loops. The first loop is over i=[0,n] and the second loop is over j=[i,n] i.e. we only need to loop over half the n by n matrix. Is there a nice c++11 way of handling these loops?

Reply:
OK, this probably remains the best approach for now.